x-pack/plugins/ml/public/application/components/job_spaces_repair/repair_results.tsx

```diff
@@ -0,0 +1,178 @@
+        <h3>
+          <EuiTextColor color={items.length ? 'default' : 'subdued'}>
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.savedObjectsDeleted.title"
+              defaultMessage="Unmatched job saved objects ({count})"
+              values={{ count: items.length }}
+            />
+          </EuiTextColor>
+        </h3>
+      </EuiTitle>
+      <EuiText size="s">
+        <p>
+          <EuiTextColor color="subdued">
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.savedObjectsDeleted.description"
+              defaultMessage="Saved objects which do not have an accompanying job. These will be left over when deleting anomaly detection jobs."
+            />
+          </EuiTextColor>
+        </p>
+      </EuiText>
+    </>
+  );
+  return <RepairItem id="savedObjectsDeleted" title={title} items={items} />;
+};
+
+const DatafeedsAdded: FC<{ repairItems: RepairSavedObjectResponse }> = ({ repairItems }) => {
+  const items = Object.keys(repairItems.datafeedsAdded);
+
+  const title = (
+    <>
+      <EuiTitle size="xs">
+        <h3>
+          <EuiTextColor color={items.length ? 'default' : 'subdued'}>
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.datafeedsAdded.title"
+              defaultMessage="Job saved objects with missing datafeeds ({count})"
+              values={{ count: items.length }}
+            />
+          </EuiTextColor>
+        </h3>
+      </EuiTitle>
+      <EuiText size="s">
+        <p>
+          <EuiTextColor color="subdued">
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.datafeedsAdded.description"
+              defaultMessage="Anomaly detection job saved objects which are missing their datafeed id."
+            />
+          </EuiTextColor>
+        </p>
+      </EuiText>
+    </>
+  );
+  return <RepairItem id="datafeedsAdded" title={title} items={items} />;
+};
+
+const DatafeedsRemoved: FC<{ repairItems: RepairSavedObjectResponse }> = ({ repairItems }) => {
+  const items = Object.keys(repairItems.datafeedsRemoved);
+
+  const title = (
+    <>
+      <EuiTitle size="xs">
+        <h3>
+          <EuiTextColor color={items.length ? 'default' : 'subdued'}>
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.datafeedsRemoved.title"
+              defaultMessage="Job saved objects with unmatched datafeed ids ({count})"
+              values={{ count: items.length }}
+            />
+          </EuiTextColor>
+        </h3>
+      </EuiTitle>
+      <EuiText size="s">
+        <p>
+          <EuiTextColor color="subdued">
+            <FormattedMessage
+              id="xpack.ml.management.repairSavedObjectsFlyout.datafeedsRemoved.description"
+              defaultMessage="Anomaly detection jobs which contain a datafeed ID, but the datafeed does not exist."
```

Comment:
It would be nice to follow the style of the other messages and describe what will happen in this circumstance. For example:
```suggestion
              defaultMessage="If there are saved objects that use a datafeed that does not exist, they will be deleted."
```

Reply:
updated in https://github.com/elastic/kibana/pull/83320/commits/29c6175bd0c1efa92d388f57dfb34a8fcb23e47e